### ISSUE TITLE
Allow importing glTF at runtime with colliders

### ISSUE THREAD

**Issue by Cretezy** [godotengine/godot-proposals#5268]
### Describe the project you are working on

A racing game with dynamic maps. Goal is to allow users to upload maps as glTF and have others play them (dynamic model loading).

### Describe the problem or limitation you are having in your project

When importing glTF at runtime, the mesh is created, but the colliders (using `-col` or similar) are not generated.

```gdscript
var gstate = GLTFState.new()
var gltf = GLTFDocument.new()

gltf.append_from_file("res://Map.glb", gstate, 0, 30)

var scene = gltf.generate_scene(gstate)
add_child(scene)
```

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow runtime generation of the colliders.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Having a flag for the `append_from_file` and related methods to allow importing of colliders would be ideal.

```
gltf.append_from_file("res://Map.glb", gstate, SOME_FLAG_FOR_COLLIDER_CREATION, 30)
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Have not found any way to work around it in the runtime. Only thing I can think of is compiling the map from glTF to a scene using an editor build, but I don't know how to do that.

Edit: @jgillich has provided a workout by using `create_convex_collision`/`create_trimesh_collision`, but this requires re-implementing the importer's collider creator manually.

### Is there a reason why this should be core and not an add-on in the asset library?

Core already contains the code to generate colliders as part of the import process.

**Comment by Cretezy** [comment]
Additional implementation details:

Look like the `ResourceImporterScene::_pre_fix_node` function is what does the check for `col`, called from `ResourceImporterScene::import`.

**Comment by jgillich** [comment]
You can generate colliders using `create_convex_collision` and `create_trimesh_collision` on MeshInstance: https://docs.godotengine.org/en/latest/classes/class_meshinstance3d.html#class-meshinstance3d-method-create-convex-collision

Not sure if this produces the exact same results as the importer though.

**Comment by Cretezy** [comment]
@jgillich It looks like that could work as a workout (I haven't tried it).

This solution would require me to re-implement the code for the importer though. Ideally, the loader would have the option to run the engine-provided collider creation at the same time.

**Comment by fire** [comment]
We discuss methods of solving this.

1. Encode a way to turn the syntax "_colliders" and have a way to do this for every type of gltf importer
2. Make a gltf collider and physics extension. https://github.com/omigroup/gltf-extensions/pull/75 https://github.com/omigroup/gltf-extensions/pull/63

**Comment by Cretezy** [comment]
@fire Could you explain in simpler terms? I'm not very familiar with the presented options. Thank you!

**Comment by fire** [comment]
Someone has to define a standard for saving colliders into gltfs and then write a plugin for it.

**Comment by Cretezy** [comment]
@fire Got it. So this would be a departure of import hints? (https://docs.godotengine.org/en/stable/tutorials/assets_pipeline/importing_scenes.html#import-hints)

**Comment by TokageItLab** [comment]
I remember discussing with @fire earlier how to handle retarget at runtime as well. Do we need an implementation to run post importer at runtime? Or would it work if the .import file was generated before the runtime load?

**Comment by aaronfranke** [comment]
Support for specifying physics information explicitly in a GLTF file via OMI physics, including primitive shapes, mesh-based shapes, and various body types, is implemented in this PR: https://github.com/godotengine/godot/pull/69266

Please review, let me know if this works for your use cases, if you find any bugs, and if you have any feedback.

**Comment by aaronfranke** [comment]
Implemented in https://github.com/godotengine/godot/pull/69266 and its follow-up PRs https://github.com/godotengine/godot/pull/77602, https://github.com/godotengine/godot/pull/77938, https://github.com/godotengine/godot/pull/78352, https://github.com/godotengine/godot/pull/78967, https://github.com/godotengine/godot/pull/88301, and maybe more. I'm going to change the milestone on this proposal to Godot 4.3 because that is when all of these follow-ups are merged.
